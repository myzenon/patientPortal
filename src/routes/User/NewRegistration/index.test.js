import React from 'react'
import { NewRegistration } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test('renders correctly', () => {
    const wrapper = render (
        <NewRegistration />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test('clicks next button | found = true | accountExists = false', (done) => {
    const props = {
        push: (url) => {
            expect(url).toBe('/new-registration-hpid-link-confirm')
            done()
        },
        callSyncAPIGateway: (name, get, post) => {
            expect(name).toBe('usersLookupInsuranceIdPost')
            expect(post).toEqual({
                insurance_id: 'INSURANCEID'
            })
            return Promise.resolve()
        },
        page: {
            apiData: {
                found: 'true',
                accountExists: 'false'
            }
        },
        savePageData: (data) => {
            expect(data).toEqual({
                insuranceId: 'INSURANCEID'
            })
            return Promise.resolve()
        }
    }
    const wrapper = mount (
        <NewRegistration {...props} />
    )
    wrapper.find('input[name="insuranceId"]').simulate('change', {
        target: {
            name: 'insuranceId',
            value: 'insuranceId'
        }
    })
    wrapper.find('button[type="submit"]').simulate('submit')
})

test('clicks next button | found = true | accountExists = true', (done) => {
    const props = {
        callSyncAPIGateway: (name, get, post) => {
            expect(name).toBe('usersLookupInsuranceIdPost')
            expect(post).toEqual({
                insurance_id: 'INSURANCEID'
            })
            return Promise.resolve()
        },
        page: {
            apiData: {
                found: 'true',
                accountExists: 'true'
            }
        },
        savePageData: (data) => {
            expect(data).toEqual({
                insuranceId: 'INSURANCEID'
            })
            return Promise.resolve()
        }
    }
    const wrapper = mount (
        <NewRegistration {...props} />
    )
    wrapper.find('input[name="insuranceId"]').simulate('change', {
        target: {
            name: 'insuranceId',
            value: 'insuranceId'
        }
    })
    wrapper.find('button[type="submit"]').simulate('submit')
    setTimeout(function () {
        expect(toJson(wrapper.render())).toMatchSnapshot()
        done()
    }, 0)
})

test('clicks next button | found = false', (done) => {
    const props = {
        callSyncAPIGateway: (name, get, post) => {
            expect(name).toBe('usersLookupInsuranceIdPost')
            expect(post).toEqual({
                insurance_id: 'INSURANCEID'
            })
            return Promise.resolve()
        },
        page: {
            apiData: {
                found: 'false'
            }
        },
        savePageData: (data) => {
            expect(data).toEqual({
                insuranceId: 'INSURANCEID'
            })
            return Promise.resolve()
        }
    }
    const wrapper = mount (
        <NewRegistration {...props} />
    )
    wrapper.find('input[name="insuranceId"]').simulate('change', {
        target: {
            name: 'insuranceId',
            value: 'insuranceId'
        }
    })
    wrapper.find('button[type="submit"]').simulate('submit')
    setTimeout(function () {
        expect(toJson(wrapper.render())).toMatchSnapshot()
        done()
    }, 0)
})
