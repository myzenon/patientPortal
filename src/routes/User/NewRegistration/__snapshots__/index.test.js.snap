// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicks next button | found = false 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <div
      class="instructionWrapper"
    >
      <p>
        To get started, please enter your
      </p>
      <p>
        Health Plan Identification Number below
      </p>
    </div>
    <form
      class=""
    >
      <div
        class="inputTextWrapper"
      >
        <input
          class="inputTextCenterError"
          name="insuranceId"
          placeholder="Enter Health Plan Identification Number"
          textalign="center"
          type="text"
          value="insuranceId"
        />
        <div
          class="textError"
        >
          Uh oh! It looks like you might have entered the wrong number, or we can't find you in our system. Give us a call  and we'll help you sort it out!
        </div>
      </div>
      <br />
       
      <br />
      <p />
      <div
        class="helpWrapper"
      >
        <a
          href="/help-me-find-hpid"
        >
          <br />
           Help me find my Health Plan Identification Number
        </a>
      </div>
      <br />
      <p />
      <button
        class="button btn btn-secondary"
        type="submit"
      >
        <b>
          NEXT
        </b>
      </button>
    </form>
    <div
      class="questionWrapper"
    />
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepActive"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="step"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumber"
          >
            2
          </div>
        </div>
        <div
          class="step"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumber"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`clicks next button | found = true | accountExists = true 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <div
      class="instructionWrapper"
    >
      <p>
        To get started, please enter your
      </p>
      <p>
        Health Plan Identification Number below
      </p>
    </div>
    <form
      class=""
    >
      <div
        class="inputTextWrapper"
      >
        <input
          class="inputTextCenterError"
          name="insuranceId"
          placeholder="Enter Health Plan Identification Number"
          textalign="center"
          type="text"
          value="insuranceId"
        />
        <div
          class="textError"
        >
          Uh oh! It looks like this insurance id is already associated with an account.Â 
          <a
            href="/login"
          >
            Click here to Sign In.
          </a>
        </div>
      </div>
      <br />
       
      <br />
      <p />
      <div
        class="helpWrapper"
      >
        <a
          href="/help-me-find-hpid"
        >
          <br />
           Help me find my Health Plan Identification Number
        </a>
      </div>
      <br />
      <p />
      <button
        class="button btn btn-secondary"
        type="submit"
      >
        <b>
          NEXT
        </b>
      </button>
    </form>
    <div
      class="questionWrapper"
    />
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepActive"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="step"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumber"
          >
            2
          </div>
        </div>
        <div
          class="step"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumber"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <div
      class="instructionWrapper"
    >
      <p>
        To get started, please enter your
      </p>
      <p>
        Health Plan Identification Number below
      </p>
    </div>
    <form
      class=""
    >
      <div
        class="inputTextWrapper"
      >
        <input
          autofocus=""
          class="inputTextCenter"
          name="insuranceId"
          placeholder="Enter Health Plan Identification Number"
          textalign="center"
          type="text"
          value=""
        />
      </div>
      <br />
       
      <br />
      <p />
      <div
        class="helpWrapper"
      >
        <a
          href="/help-me-find-hpid"
        >
          <br />
           Help me find my Health Plan Identification Number
        </a>
      </div>
      <br />
      <p />
      <button
        class="button btn btn-secondary"
        type="submit"
      >
        <b>
          NEXT
        </b>
      </button>
    </form>
    <div
      class="questionWrapper"
    />
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepActive"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="step"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumber"
          >
            2
          </div>
        </div>
        <div
          class="step"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumber"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
